nilearn/maskers/surface_labels_masker.py

```diff
@@ -307,15 +307,14 @@ def _fit(self, imgs):
         else:
             self.clean_args_ = self.clean_args
 
-        if not self.reports:
-            return self
-
-        for part in self.labels_img_.data.parts:
-            self._report_content["n_vertices"][part] = (
-                self.labels_img_.mesh.parts[part].n_vertices
-            )
+        if self.reports:
+            self._report_content["reports_at_fit_time"] = True
+            for part in self.labels_img_.data.parts:
+                self._report_content["n_vertices"][part] = (
+                    self.labels_img_.mesh.parts[part].n_vertices
+                )
```

Comment:
No change actually in the code, just changed the if condition and indentation. This way, reports check is standard in all maskers and `mask_logger` line below is executed.